1.illumination_correction/scripts/0.create_loaddata_csvs.py

```diff
@@ -46,16 +46,31 @@
     # Get the first folder directly under the index_directory
     relative_path = folder.relative_to(index_directory)
     first_folder = relative_path.parts[0]  # First-level folder
-
-    # Generate the plate name based on folder structure
+    
+    # Generate the plate name and find matching config file based on folder structure
     if first_folder.startswith('BR00'):
         plate_name = first_folder.split('_')[0]  # Take the first part
+        config_path = config_dir_path / "config.yml"  # Use default config for BR00
 
     elif first_folder.startswith('2024'):
         second_folder = relative_path.parts[1]  # Second-level folder
-        part1 = '_'.join(first_folder.split('_')[-2:])  # Last two parts of first folder
-        part2 = second_folder.split('_')[0]  # First part of second folder
-        plate_name = f"{part1}_{part2}"  # Combine them
+        part1 = '_'.join(first_folder.split('_')[-2:])  # Last two parts of first folder ("CellLine_Re-imaged")
+        part2 = second_folder.split('_')[0]  # First part of second folder (BR00 ID)
+
+        # Combine to make plate name for saving CSV
+        plate_name = f"{part1}_{part2}" 
+
+        # Find the matching config file by matching part1 with the config file's prefix
+        matching_configs = list(config_dir_path.glob(f"{part1.split('_')[0]}_*.yml"))
+        
+        # Debugging print to check if matching config files are found
+        print(f"Matching configs found: {matching_configs}")
+
+        if matching_configs:
+            config_path = matching_configs[0]  # Take the first match
+        else:
+            print(f"No matching config file for: {part1}")
+            continue  # Skip if no matching config file
```

Comment:
would it be more accurate to say something like: "Skip and use default config file if no specific config found")?

Reply:
In this case it is not because we just skip processing that data entirely, which in this case since we are having issues with `ChannelID`s changing, it might actually be better to skip and report than assign a config file which might mess up the data.